Enable multicore samples for imx943 evk

biwenli opened the pull request:
Enable multicore samples between mcus(m33s in NETCMIX, M7_0 in M7MIX0, M7_1 in M7MIX1) for imx943 evk

uLipe commented on boards/nxp/imx943_evk/imx943_evk_mimx94398_m33.dts:
```diff
@@ -18,8 +18,10 @@
+		zephyr,console = &lpuart3;
+		zephyr,shell-uart = &lpuart3;
+		//zephyr,console = &lpuart8;
+		//zephyr,shell-uart = &lpuart8;
```
Is there any special reason to make these commented rather than removed?

uLipe commented on boards/nxp/imx943_evk/imx943_evk_mimx94398_m33.dts:
```diff
@@ -105,7 +107,20 @@
+/* GND(J45-12)                               --- GND of usb2serial converter */
+&lpuart3 {
+	status = "disabled";
+	//status = "okay";
```
Also this

biwenli replied:
> Also this

For debugging.

uLipe commented on boards/nxp/imx943_evk/imx943_evk_mimx94398_m33.dts:
```diff
@@ -105,7 +107,20 @@
+};
+
 &lpuart8 {
+	//status = "disabled";
```
Also this.

uLipe commented on boards/nxp/imx943_evk/imx943_evk_mimx94398_m7_0.dts:
```diff
@@ -0,0 +1,47 @@
+	compatible = "nxp,imx943_evk";
+
+	chosen {
+		/* TCM */
```
This comment may not be necessary, since lines below are self explanatory

biwenli replied:
> This comment may not be necessary, since lines below are self explanatory

Sure, will delete it.

uLipe commented on boards/nxp/imx943_evk/imx943_evk_mimx94398_m7_1.dts:
```diff
@@ -0,0 +1,47 @@
+	compatible = "nxp,imx943_evk";
+
+	chosen {
+		/* TCM */
```
Also this.

uLipe reviewed (commented):
Nice work! I have some nits.

dleach02 commented:
@biwenli can you address the merge conflicts

biwenli commented on boards/nxp/imx943_evk/imx943_evk_mimx94398_m33.dts:
```diff
@@ -14,12 +14,14 @@
+		/* Note: please enable lpuart3 and disable lpuart8 for debugging via JTAG */
 		zephyr,console = &lpuart8;
 		zephyr,shell-uart = &lpuart8;
+		//zephyr,console = &lpuart3;
```
> Please remove commented code - here and in the rest of this PR. Otherwise, looks good.

sure, np.

dbaluta commented on boards/nxp/imx943_evk/doc/index.rst:
```diff
@@ -375,9 +375,9 @@ For DDR target
+a. Please connect two additional USB-to-Serial converters between the Host PC and the board's
+Arduino interface using Dupont cables. For M70 in M70 MIX and M71 in M71 MIX,
+make the connections as shown below.
 
```
This doesn't belong to this patch. Each patch should be a single logical change. This is a spell fix and needs to be placed in a separate patch.

dbaluta commented on samples/subsys/ipc/ipc_service/static_vrings/boards/imx943_evk_mimx94398_m7_1.overlay:
```diff
@@ -0,0 +1,51 @@
+		//	mbox-names = "tx", "rx";
+		//	role = "host";
+		//	status = "okay";
+		//};
```
Pleae do not add commented code. If the code is not used should be deleted. If the code serves as a documentation should be placed in documentation section.

biwenli replied:
> Pleae do not add commented code. If the code is not used should be deleted. If the code serves as a documentation should be placed in documentation section.

okay, got it.

iuliana-prodan commented on samples/drivers/mbox/boards/imx943_evk_mimx94398_m33.overlay:
```diff
@@ -0,0 +1,17 @@
+		 * m33s(NETCMIX) as master(host), m71(M7MIX1) as remote
+		 */
+		mboxes = <&mu_m33s_m71_for_m33s 1>, <&mu_m33s_m71_for_m33s 0>;
+		//mboxes = <&mu_m33s_m70_for_m33s 1>, <&mu_m33s_m70_for_m33s 0>;
```
remove commented core.

iuliana-prodan commented on samples/drivers/mbox_data/boards/imx943_evk_mimx94398_m33.overlay:
```diff
@@ -0,0 +1,17 @@
+		 * m33s(NETCMIX) as master(host), m71(M7MIX1) as remote
+		 */
+		mboxes = <&mu_m33s_m71_for_m33s 1>, <&mu_m33s_m71_for_m33s 0>;
+		//mboxes = <&mu_m33s_m70_for_m33s 1>, <&mu_m33s_m70_for_m33s 0>;
```
remove commented core.

iuliana-prodan commented on samples/drivers/mbox_data/boards/imx943_evk_mimx94398_m7_0.overlay:
```diff
@@ -0,0 +1,17 @@
+		 * m70(M7MIX0) as master(host), m33s(NETCMIX) as remote
+		 */
+		mboxes = <&mu_m33s_m70_for_m70 1>, <&mu_m33s_m70_for_m70 0>;
+		//mboxes = <&mu_m70_m71_for_m70 1>, <&mu_m70_m71_for_m70 0>;
```
remove commented core.

iuliana-prodan commented on samples/drivers/mbox_data/boards/imx943_evk_mimx94398_m7_1.overlay:
```diff
@@ -0,0 +1,17 @@
+		 * m71(M7MIX1) as master(host), m33s(NETCMIX) as remote
+		 */
+		mboxes = <&mu_m33s_m71_for_m71 1>, <&mu_m33s_m71_for_m71 0>;
+		//mboxes = <&mu_m70_m71_for_m71 1>, <&mu_m70_m71_for_m71 0>;
```
remove commented core.

iuliana-prodan commented on samples/subsys/ipc/ipc_service/static_vrings/boards/imx943_evk_mimx94398_m7_0.overlay:
```diff
@@ -0,0 +1,52 @@
+		//	status = "okay";
+		//};
+
+		//ipc1: ipc1 {
```
remove commented core.

iuliana-prodan commented on samples/subsys/ipc/ipc_service/static_vrings/remote/boards/imx943_evk_mimx94398_m33.overlay:
```diff
@@ -0,0 +1,53 @@
+			status = "okay";
+		};
+
+		//ipc0: ipc0 {
```
remove commented core.

iuliana-prodan commented on samples/subsys/ipc/ipc_service/static_vrings/remote/boards/imx943_evk_mimx94398_m33.overlay:
```diff
@@ -0,0 +1,53 @@
+		//	status = "okay";
+		//};
+
+		//ipc1: ipc1 {
```
remove commented core.

iuliana-prodan commented on samples/subsys/ipc/ipc_service/static_vrings/remote/boards/imx943_evk_mimx94398_m7_0.overlay:
```diff
@@ -0,0 +1,55 @@
+		//	status = "okay";
+		//};
+
+		//ipc1: ipc1 {
```
remove commented core.

iuliana-prodan commented on samples/subsys/ipc/ipc_service/static_vrings/remote/boards/imx943_evk_mimx94398_m7_1.overlay:
```diff
@@ -0,0 +1,55 @@
+			status = "okay";
+		};
+
+		//ipc0: ipc0 {
```
remove commented core.

iuliana-prodan commented on samples/subsys/ipc/ipc_service/static_vrings/remote/boards/imx943_evk_mimx94398_m7_1.overlay:
```diff
@@ -0,0 +1,55 @@
+		//	status = "okay";
+		//};
+
+		//ipc1: ipc1 {
```
remove commented core.

iuliana-prodan commented on samples/subsys/ipc/openamp/boards/imx943_evk_mimx94398_m33.overlay:
```diff
@@ -0,0 +1,22 @@
+
+		zephyr,ipc_shm = &dram_m33s_m71_sh_mem;
+		zephyr,ipc = &mailbox_m33s_m71_for_m33s_as_master;
+		//zephyr,ipc_shm = &dram_m33s_m70_sh_mem;
```
remove commented core.

iuliana-prodan commented on samples/subsys/ipc/openamp/remote/boards/imx943_evk_mimx94398_m33.overlay:
```diff
@@ -0,0 +1,21 @@
+
+		zephyr,ipc_shm = &dram_m33s_m71_sh_mem;
+		zephyr,ipc = &mailbox_m33s_m71_for_m33s_as_remote;
+		//zephyr,ipc_shm = &dram_m33s_m70_sh_mem;
```
remove commented core.

iuliana-prodan reviewed (changes requested):
There’s _still_ a lot of commented-out code - either remove it or add an explanation for why it’s there.
If you need to explain how a user should switch between m7_0 and m7_1, consider adding that information in a README file.

biwenli commented on boards/nxp/imx943_evk/imx943_evk_mimx94398_m33.dts:
```diff
@@ -107,7 +107,20 @@
 
+/* Note: Please disable lpuart8 when debugging with lpuart3 */
+/* GPIO_IO30(J51-18, M1_PWM_CX, LPUART3_TX)  --> RX of usb2serial converter  */
+/* GPIO_IO31(J44-10, M1_LED_TP1, LPUART3_RX) <-- TX of usb2serial converter  */
```
> Can you please enable each sample in its own commit with short description of why the change is needed and the benefits.

okay,

biwenli commented on boards/nxp/imx943_evk/imx943_evk_mimx94398_m33.dts:
```diff
@@ -107,7 +107,20 @@
+/* GND(J45-12)                               --- GND of usb2serial converter */
+&lpuart3 {
+	status = "disabled";
+	//status = "okay";
```
> remove commented core.

sure, np.

biwenli commented on samples/drivers/mbox/boards/imx943_evk_mimx94398_m7_0.overlay:
```diff
@@ -0,0 +1,17 @@
+		 * m70(M7MIX0) as master(host), m33s(NETCMIX) as remote
+		 */
+		mboxes = <&mu_m33s_m70_for_m70 1>, <&mu_m33s_m70_for_m70 0>;
+		//mboxes = <&mu_m70_m71_for_m70 1>, <&mu_m70_m71_for_m70 0>;
```
> remove commented core.

sure, np.

biwenli commented on samples/subsys/ipc/openamp/boards/imx943_evk_mimx94398_m7_0.overlay:
```diff
@@ -0,0 +1,21 @@
+
+		zephyr,ipc_shm = &dram_m33s_m70_sh_mem;
+		zephyr,ipc = &mailbox_m33s_m70_for_m70_as_master;
+		//zephyr,ipc_shm = &dram_m70_m71_sh_mem;
```
> remove commented core.

sure, np.

biwenli committed:
boards: nxp: imx943_evk: doc: spell fix

This commit fixes spell and add info
to debugging system via JTAG interface.

Signed-off-by: Biwen Li <biwen.li@nxp.com>

biwenli committed:
boards: nxp: imx943_evk: m33: add lpuart3 node for debugging

Add lpuart3 node for debugging via JTAG interface

Signed-off-by: Biwen Li <biwen.li@nxp.com>

JiafeiPan commented on dts/arm/nxp/nxp_imx94x.dtsi:
```diff
@@ -342,7 +568,181 @@
+			status = "disabled";
+		};
+	};
+	/* Define memory regions for IPC between m33s and m71 */
```
For this and all the following IPC memory regions which are in DDR memory, per my understand they should could be customized, right? if yes, I suggest move them to board dts because they are not fixed value in SoC, they should be board level definitions according to external memory resource available.

biwenli replied:
> For this and all the following IPC memory regions which are in DDR memory, per my understand they should could be customized, right? if yes, I suggest move them to board dts because they are not fixed value in SoC, they should be board level definitions according to external memory resource available.

There memory addresses are fixed value. If the board want to use different address. we can over write it in board level.

JarmouniA reviewed (changes requested):
Please split into multiple commits as follows:
- SoC dts changes (1 commit per SoC)
- Board changes
- 1 commit per Sample

biwenli commented:
> Please split into multiple commits as follows:
> 
> * SoC dts changes (1 commit per SoC)
> * Board changes
> * 1 commit per Sample

sure, np.

biwenli committed:
dts: arm: nxp: imx943: add mu and mbox nodes

This commit adds mu and mbox nodes
for Inter Processor Communication
and reuse nodes.

Signed-off-by: Biwen Li <biwen.li@nxp.com>

biwenli committed:
boards: nxp: imx943_evk: remove robust comments

Remove robust comments.

Signed-off-by: Biwen Li <biwen.li@nxp.com>

biwenli committed:
samples: enable mbox sample for imx943 evk

Enable mbox sample for imx943 evk.

Default configuration in mbox sample(as master):
m33 as master, m7_1 as remote.
m7_0 as master, m33 as remote.
m7_1 as master, m33 as remote.

Default configuration in mbox/remote sample(as remote):
m7_1 as master, m33 as remote.
m33 as master, m7_0 as remote.
m33 as master, m7_1 as remote.

Signed-off-by: Biwen Li <biwen.li@nxp.com>

biwenli committed:
samples: enable mbox_data sample for imx943 evk

Enable mbox_data sample for imx943 evk.

Default configuration in mbox_data sample(as master):
m33 as master, m7_1 as remote.
m7_0 as master, m33 as remote.
m7_1 as master, m33 as remote.

Default configuration in mbox_data/remote sample(as remote):
m7_1 as master, m33 as remote.
m33 as master, m7_0 as remote.
m33 as master, m7_1 as remote.

Signed-off-by: Biwen Li <biwen.li@nxp.com>

biwenli committed:
samples: enable ipc static vrings for imx943 evk

Enable ipc static vrings sample for imx943 evk.

Default configuration in static_vrings sample(as master):
m33 as master, m7_1 as remote.
m7_0 as master, m33 as remote.
m7_1 as master, m33 as remote.

Default configuration in static_vrings/remote sample(as remote):
m7_1 as master, m33 as remote.
m33 as master, m7_0 as remote.
m33 as master, m7_1 as remote.

Signed-off-by: Biwen Li <biwen.li@nxp.com>

biwenli committed:
samples: enable openamp sample for imx943 evk

Enable openamp sample for imx943 evk.

Default configuration in openamp sample(as master):
m33 as master, m7_1 as remote.
m7_0 as master, m33 as remote.
m7_1 as master, m33 as remote.

Default configuration in openamp/remote sample(as remote):
m7_1 as master, m33 as remote.
m33 as master, m7_0 as remote.
m33 as master, m7_1 as remote.

Signed-off-by: Biwen Li <biwen.li@nxp.com>

kartben commented:
> @JiafeiPan need your review here

@decsny they have approved - GitHub UI is just messy these days

<img width="444" height="326" alt="image" src="https://github.com/user-attachments/assets/dcee634b-188b-4adb-a88f-bbbdb8d53f0e" />
